shelljs/shelljs.d.ts

```diff
@@ -434,31 +436,44 @@ declare module "shelljs"
      * Object containing environment variables (both getter and setter). Shortcut to process.env.
      */
     export var env: { [key: string]: string };
-
-    /*
-
-    // Not yet implemented due to implementation issues (constant overloads and return types).
-    // See: https://github.com/arturadib/shelljs#execcommand--options--callback
-
-    export function exec(command: string, options: ExecOptions, callback: (code: number, output: string) => any): any;
-    export function exec(command: string, options: ExecOptions): any;
-
-    interface ExecOptions
-    {
-        silent: boolean;
-        async: boolean;
-    }
-
-    */
-
+    
     /**
      * Executes the given command synchronously.
-     * @param  {string}          command The commadn to execute.
+     * @param  {string}          command The command to execute.
      * @return {ExecReturnValue}         Returns an object containing the return code and output as string.
      */
-    export function exec(command: string): ExecReturnValue;
+    export function exec(command: string): ExecOutputReturnValue;
+    /**
+     * Executes the given command synchronously.
+     * @param  {string}          command The command to execute.
+     * @param  {ExecOptions}     options Silence and synchronous options.
+     * @return {ExecReturnValue}         Returns an object containing the return code and output as string.
+     */
+    export function exec(command: string, options: ExecOptions): ExecOutputReturnValue;
```

Comment:
Well, if the return type is `any`, you have to cast anyway to access a member. So why not tighten this a little bit?

Just played with this a little bit using the playground. This seems to work as expected ([link](http://www.typescriptlang.org/Playground#src=interface%20ReturnValue%0D%0A{%0D%0A%09output%3A%20string%3B%09%0D%0A}%0D%0A%0D%0Ainterface%20ChildProcess%0D%0A{%0D%0A%09id%3A%20number%3B%09%0D%0A}%0D%0A%0D%0Afunction%20a%28%29%3A%20any%0D%0A{%0D%0A%09if%20%28Math.random%28%29%20%3E%200.5%29%0D%0A%09%09return%20{%20output%3A%20%22lol%22%20}%3B%0D%0A%09return%20{%20id%3A%202%20}%3B%0D%0A}%0D%0Afunction%20b%28%29%3A%20ReturnValue|ChildProcess%0D%0A{%0D%0A%09if%20%28Math.random%28%29%20%3E%200.5%29%0D%0A%09%09return%20{%20output%3A%20%22lol%22%20}%3B%0D%0A%09return%20{%20id%3A%202%20}%3B%0D%0A}%0D%0A%0D%0Avar%20aRet0%20%3D%20%3CReturnValue%3Ea%28%29%3B%0D%0Avar%20aRet1%20%3D%20%3CChildProcess%3Ea%28%29%3B%0D%0Avar%20aRet2%20%3D%20a%28%29%3B%0D%0A%0D%0Avar%20bRet0%20%3D%20%3CReturnValue%3Eb%28%29%3B%0D%0Avar%20bRet1%20%3D%20%3CChildProcess%3Eb%28%29%3B%0D%0Avar%20bRet2%20%3D%20b%28%29%3B%0D%0A)).